src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -59,26 +59,22 @@ std::optional<size_t> getBatchSizeForNode(const IONodeDescriptor& nodeDescriptor
                                           const ZeroExecutor::ArgumentDescriptor& zeDescriptor) {
     Logger logger("GetBatchSizeForNode", Logger::global().level());
 
+    if (nodeDescriptor.originalShape.is_dynamic() || nodeDescriptor.originalShape.rank().get_length() == 0) {
+        logger.info("Dynamic networks are not supported when batching is handled by the plugin");
```

Comment:
I find the logger message a bit confusing in the `nodeDescriptor.originalShape.rank().get_length() == 0` case, maybe we should display different messages for different corner cases.

Reply:
created a separate check and message for this case. thanks!